dhall/tests/Dhall/Test/TypeInference.hs

```diff
@@ -18,9 +20,21 @@ import qualified Test.Tasty        as Tasty
 import qualified Test.Tasty.HUnit  as Tasty.HUnit
 import qualified Turtle
 
+typeInferenceDirectory :: FilePath
+typeInferenceDirectory = "./dhall-lang/tests/type-inference"
+
 getTests :: IO TestTree
 getTests = do
-    successTests <- Test.Util.discover (Turtle.chars <* "A.dhall") successTest (Turtle.lstree "./dhall-lang/tests/type-inference/success")
+    let successTestFiles = do
+            path <- Turtle.lstree (typeInferenceDirectory </> "success")
+
+            let skip = [ typeInferenceDirectory </> "success/unit/AssertNaNA.dhall" ]
```

Comment:
This test failure is due to our test setup that compares the inferred and the expected types with `==` on `Expr`. `Expr`'s `Eq` instance is derived and therefore lifts the classic `NaN /= ǸaN` inequality. Not sure whether we should change the `Eq` instance. Maybe we should document the lack of reflexivity though.

Reply:
I have added docs for now.